restapi/src/main/java/io/stargate/web/resources/v2/schemas/IndexesResource.java

```diff
@@ -0,0 +1,262 @@
+import io.stargate.auth.SourceAPI;
+import io.stargate.auth.UnauthorizedException;
+import io.stargate.db.schema.CollectionIndexingType;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.ImmutableCollectionIndexingType;
+import io.stargate.db.schema.Keyspace;
+import io.stargate.db.schema.Table;
+import io.stargate.web.models.Error;
+import io.stargate.web.models.IndexAdd;
+import io.stargate.web.models.IndexKind;
+import io.stargate.web.models.SuccessResponse;
+import io.stargate.web.resources.AuthenticatedDB;
+import io.stargate.web.resources.Db;
+import io.stargate.web.resources.RequestHandler;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.*;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+@Api(
+    produces = MediaType.APPLICATION_JSON,
+    consumes = MediaType.APPLICATION_JSON,
+    tags = {"schemas"})
+@Path("/v2/schemas/keyspaces/{keyspaceName}/indexes")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class IndexesResource {
+  @Inject private Db db;
+
+  @Timed
+  @POST
+  @ApiOperation(
+      value = "Add an index to a table's column",
+      notes = "Add an index to a single column of a table.",
+      response = SuccessResponse.class,
+      code = 201)
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 201, message = "Created", response = SuccessResponse.class),
+        @ApiResponse(code = 400, message = "Bad request", response = Error.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 403, message = "Forbidden", response = Error.class),
+        @ApiResponse(code = 409, message = "Conflict", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal Server Error", response = Error.class)
+      })
+  public Response addIndex(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
+      @ApiParam(required = true) @NotNull final IndexAdd indexAdd,
+      @Context HttpServletRequest request) {
+    return RequestHandler.handle(
+        () -> {
+          Map<String, String> allHeaders = getAllHeaders(request);
+          AuthenticatedDB authenticatedDB = db.getDataStoreForToken(token, allHeaders);
+
```

Comment:
This request needs to be authorized
https://github.com/stargate/stargate/blob/master/restapi/src/main/java/io/stargate/web/resources/v2/schemas/TablesResource.java#L335